src/backend/vulkan/src/window.rs

```diff
@@ -310,54 +310,61 @@ impl Instance {
         self.create_surface_from_vk_surface_khr(surface)
     }
 
-    #[cfg(feature = "winit")]
     #[allow(unreachable_code)]
-    pub fn create_surface(&self, window: &winit::window::Window) -> Surface {
+    pub fn create_surface(
+        &self,
+        has_handle: &impl raw_window_handle::HasRawWindowHandle,
+    ) -> Surface {
+        use raw_window_handle::RawWindowHandle;
+
         #[cfg(all(
             feature = "x11",
             unix,
             not(target_os = "android"),
             not(target_os = "macos")
         ))]
         {
-            use winit::platform::unix::WindowExtUnix;
-
-            if self.extensions.contains(&khr::WaylandSurface::name()) {
-                if let Some(display) = window.wayland_display() {
-                    let display: *mut c_void = display as *mut _;
-                    let surface: *mut c_void = window.wayland_surface().unwrap() as *mut _;
-                    return self.create_surface_from_wayland(display, surface);
+            return match has_handle.raw_window_handle() {
```

Comment:
could we refactor this into a single giant `match has_handle.raw_window_handle()`, where some of the arms are gated by `cfg` and some have conditions like `if self.extensions.contains(...)`? I think it would look much better here

Reply:
Agreed, will do.

Reply:
Agreed, will do.